### PR TITLE
Change global DNS target in a straightforward way

### PR DESCRIPTION
**Pull request by gitlawr**
Address https://github.com/rancher/rancher/issues/17982#issuecomment-472225106

Problem:
Cannot change globalDNS target to mcapp when it aims at projects, and vice versa

Solution:
Delete existing project target when mcapp target is set,
Delete existing mcapp target when try to add project target.

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: cmd/globaldns.go [hunk 1]
```diff
@@ -734,6 +734,14 @@ func globalDNSUpdate(ctx *cli.Context) error {
 			return err
 		}
 		update["multiClusterAppId"] = resource.ID
+		if len(entry.ProjectIDs) > 0 {
+			input := &managementClient.UpdateGlobalDNSTargetsInput{
+				ProjectIDs: entry.ProjectIDs,
+			}
+			if err := c.ManagementClient.GlobalDNS.ActionRemoveProjects(entry, input); err != nil {
+				return err
```
Comment (prachidamle) [inline cmd/globaldns.go]: What if the update fails, we should rollback to the previous state and set back the Projects?

File: cmd/globaldns.go [hunk 2]
```diff
@@ -870,9 +878,16 @@ func addGlobalDNSProjects(ctx *cli.Context) error {
...
+		}
+	}
+
 	input := &managementClient.UpdateGlobalDNSTargetsInput{
-		MultiClusterAppID: "",
-		ProjectIDs:        projectIDs,
+		ProjectIDs: projectIDs,
 	}
 
 	return c.ManagementClient.GlobalDNS.ActionAddProjects(entry, input)
```
Comment (prachidamle) [inline cmd/globaldns.go]: Same as above, handle error if ActionAddProjects fails?

**Comment by gitlawr** [comment]
@prachidamle Updated

**Comment by prachidamle** [comment]
thanks @gitlawr, that change looks correct

**Pull request merged**